modules/core/04-channel/keeper/packet.go

```diff
@@ -132,6 +132,25 @@ func (k Keeper) RecvPacket(
 		return errorsmod.Wrapf(types.ErrInvalidChannelState, "expected channel state to be one of [%s, %s], but got %s", types.OPEN, types.FLUSHING, channel.State)
 	}
 
+	// In the case of the channel being in FLUSHING we need to ensure that the
+	// packet sequence is < counterparty next sequence send. This is a defensive
+	// check and if the counterparty is implemented correctly, this should never abort.
+	if channel.State == types.FLUSHING {
+		counterpartyUpgrade, found := k.GetCounterpartyUpgrade(ctx, packet.GetDestPort(), packet.GetDestChannel())
+		if !found {
+			return errorsmod.Wrapf(types.ErrUpgradeNotFound, "counterparty upgrade not found: portID (%s), channelID (%s)", packet.GetDestPort(), packet.GetDestChannel())
+		}
+
+		// only error if the counterparty next sequence send is set (> 0)
+		counterpartyNextSequenceSend := counterpartyUpgrade.NextSequenceSend
+		if counterpartyNextSequenceSend != 0 && packet.GetSequence() >= counterpartyNextSequenceSend {
+			return errorsmod.Wrapf(
```

Comment:
document that this is unexpected error and that counterparty has messed up if this error is reached